catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java

```diff
@@ -242,8 +243,14 @@ public void validate(Metacard metacard) throws ValidationException {
             schematronReport = generateReport(metadata, validator);
             if (!schematronReport.isValid(suppressWarnings)) {
                 throw new SchematronValidationException("Schematron validation failed.",
-                        schematronReport.getErrors(),
-                        schematronReport.getWarnings());
+                        schematronReport.getErrors()
+                                .stream()
+                                .map(error -> sanitize(error))
+                                .collect(Collectors.toList()),
+                        schematronReport.getWarnings()
+                                .stream()
+                                .map(warning -> sanitize(warning))
+                                .collect(Collectors.toList()));
```

Comment:
`.collect()` returns a new list, it doesn't change the list.
I'm not sure how the schematronReport gets its errors and warnings, but maybe you could do `schematronReport.setErrors(schematronReport.getErrors()
 .stream().map(error -> sanitize(error)).collect(Collectors.toList()))`